libs/platform/ebpf_object.c

```diff
@@ -155,30 +158,30 @@ ebpf_object_initialize(
 void
 ebpf_object_acquire_reference(ebpf_core_object_t* object)
 {
-    ebpf_assert(object->marker == _ebpf_object_marker);
-    ebpf_assert(object->reference_count != 0);
-    ebpf_interlocked_increment_int32(&object->reference_count);
+    ebpf_assert(object->base.marker == _ebpf_object_marker);
+    ebpf_assert(object->base.reference_count != 0);
+    ebpf_interlocked_increment_int32(&object->base.reference_count);
 }
 
-_Requires_lock_held_(&_ebpf_object_tracking_list_lock) void _ebpf_object_release_reference_under_lock(
+_Requires_lock_held_(&_ebpf_object_tracking_list_lock) static void _ebpf_object_release_reference_under_lock(
     ebpf_core_object_t* object)
 {
     uint32_t new_ref_count;
 
     if (!object)
         return;
 
-    ebpf_assert(object->marker == _ebpf_object_marker);
-    ebpf_assert(object->reference_count != 0);
+    ebpf_assert(object->base.marker == _ebpf_object_marker);
+    ebpf_assert(object->base.reference_count != 0);
```

Comment:
Assert on the return from the interlocked decrement as this read is not atomic (no guarantees about what version it sees).

Reply:
Fixed here, and at other places in the code.